public/app/features/alerting/unified/components/silences/MatchersField.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { FC, Fragment } from 'react';
+import { Button, Field, Input, InlineLabel, Label, useStyles, Checkbox } from '@grafana/ui';
+import { GrafanaTheme } from '@grafana/data';
+import { css, cx } from '@emotion/css';
+import { useFormContext, useFieldArray } from 'react-hook-form';
+import { SilenceFormFields } from '../../types/silence-form';
+import { AlertLabels } from '../AlertLabels';
+
+interface Props {
+  className?: string;
+}
+
+const MatchersField: FC<Props> = ({ className }) => {
+  const styles = useStyles(getStyles);
+  const formApi = useFormContext<SilenceFormFields>();
+  const { register, getValues, setValue } = formApi;
+  const { fields: matchers = [], append, remove } = useFieldArray<SilenceFormFields>({ name: 'matchers' });
+
+  const addMatcher = () => {
+    const { matcherName, matcherValue, isRegex } = getValues();
+    append({ name: matcherName, value: matcherValue, isRegex });
+    setValue('matcherName', '');
+    setValue('matcherValue', '');
+    setValue('isRegex', false);
+  };
+
+  const onRemoveLabel = (index: number) => {
+    remove(index);
+  };
+  return (
+    <div className={cx(className, styles.wrapper)}>
+      <Label>Matchers</Label>
+      <div className={styles.matchers}>
+        <AlertLabels matchers={matchers} onRemoveLabel={onRemoveLabel} />
+      </div>
+      {/* Hide the fields from the form but they need to be registered in order to be in the form state */}
+      <div className={styles.displayNone}>
+        {matchers.map((matcher, index) => {
+          return (
+            <Fragment key={`${matcher.name}-${matcher.value}-${index}`}>
```

Comment:
when using `useFieldArray`, should use `matcher.id` as `key`. the `id` is provided by `useFieldArray`